ev3dev/core.py

```diff
@@ -955,6 +958,94 @@ def wait_while(self, s, timeout=None):
         """
         return self.wait(lambda state: s not in state, timeout)
 
+    def set_power(self, power):
+        self.speed_sp = power
+
+    def set_position_rotations(self, power, rotations):
+        if power > 0:
+            self.position_sp = self.position + int(rotations * self.count_per_rot)
+        else:
+            self.position_sp = self.position - int(rotations * self.count_per_rot)
+
+    def set_position_degrees(self, power, degrees):
+        rotations = float(degrees / self.count_per_rot)
+        self.set_position_rotations(power, rotations)
+
+    def set_brake(self, brake):
+        if brake:
+            self.stop_action = self.STOP_ACTION_BRAKE
+        else:
+            self.stop_action = self.STOP_ACTION_COAST
+
+    def set_time(self, ms):
+        self.time_sp = ms
+
+    def on_for_rotations(self, power, rotations, brake):
+        """
+        Rotate the motor at 'power' for 'rotations'
+        """
+        self.set_power(power)
+        self.set_position_rotations(power, rotations)
+        self.set_brake(brake)
+        self.run_to_abs_pos()
+
+    def on_for_degrees(self, power, degrees, brake):
+        """
+        Rotate the motor at 'power' for 'degrees'
+        """
+        self.set_power(power)
+        self.set_position_degrees(power, degrees)
+        self.set_brake(brake)
+        self.run_to_abs_pos()
+
+    def on_for_seconds(self, power, seconds, brake):
```

Comment:
All of the `on_for_*` methods should wait for completion before returning.

e.g., this method should not use `self.run_time()`, it should

    self.run_forever()
    time.sleep(seconds)
    self.stop()

Reply:
On second thought, my suggestion would not be a good implementation when there are multiple threads.

Unfortunately, we don't currently have a way to be notified by the motor driver that the current operation was interrupted.

Reply:
We could have a `wait_until_stopped` arg and if it is True we call `self.wait_until_not_moving()`.  But then our API would no longer match the EV3-G API exactly.  We could just document how to use `self.wait_until_not_moving()`

Reply:
Why can't we wait until `'running' not in self.state`? Does the driver consider itself "running" while holding?

Reply:
In EV3-G if you tell a motor to run for 3 rotations does it start the motor and move on to the next block or does it wait for the three rotations to finish?  If it moves on then I think we are good with what we have now, it if blocks though then we need to add something like
```
    self.wait_until('running', timeout=100)
    self.wait_until_not_moving()
```

Reply:
It's a blocking call. It starts the motors and only returns to the caller once the motor has turned to the specified position. EV3-G programs are very commonly just a single sequence of motor blocks with some branching (drive forward, turn, drive forward, check sensor, etc.).

Reply:
ok lets have it block by default but have an optional arg that can be passed to not block